src/shell/components/global-instance/GlobalInstance.js

```diff
@@ -100,6 +109,30 @@ export default connect(state => {
             </li>
           ))}
         </ul>
+
+        {canPurge && (
+          <div>
+            <Button
+              kind="warn"
+              disabled={purge}
+              onClick={() => {
+                setPurge(true);
+                return request(
+                  `${CONFIG.CLOUD_FUNCTIONS_DOMAIN}/fastlyPurge?zuid=${props.instance.ZUID}`
+                ).then(res => {
```

Comment:
use the `finally` function, so it stops on failure. add a `catch` and dispatch a notice if it failed.